### PR TITLE
Feat/shop and card

### PR DESCRIPTION
**Pull request by arsamhb**

### CODE REVIEW

File: front/src/pages/Store/Shop/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          ? "purchased"
+          : item === "selected"
+          ? "available"
+          : "selected"
+        : item
+    );
+    setItemsAvailabilityList(newStatus);
+  };
+
+  const addToCart = (itemId: string, index: number) => {
```
[the remark this answers is not in the document]
Comment (AKA-Abdol) [reply front/src/pages/Store/Shop/index.tsx:44]: we can push some critical api calls to the stateful component of course card.

File: front/src/pages/Store/Shop/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        <div className="w-full h-[100vh] flex items-center justify-center px-12">
+          <Link to="/auth" className=" text-center text-[#58423A]">
+            برای خرید دوره ها ابتدا باید
+            <span className=" underline text-[#D48B71]"> وارد </span> حساب
+            کاربری خود شوید.
+          </Link>
+        </div>
+      ) : (
+        <div className="pt-28 mx-8 justify-center items-center flex flex-col gap-y-4 ">
+          {shopData?.courses.map((item, index) => (
```
Comment (AKA-Abdol) [inline front/src/pages/Store/Shop/index.tsx:92]: use isLoading for this.

File: front/src/App.tsx [hunk 3]
```diff
@@ -36,7 +38,11 @@ function App() {
             <Route path="contact" element={<Contact />} />
             <Route path="terms" element={<Terms />} />
             <Route path="ticket" element={<TicketForm />} />
-            <Route path="store" element={<Store />} />
+
+            <Route path="Store/" element={<Store />}>
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/App.tsx:42]: done

File: front/src/components/Header.tsx [hunk 4]
```diff
@@ -6,10 +6,10 @@ import { Link } from "react-router-dom";
 export default function Header() {
   const [menuState, setMenuState] = useState(false);
   const [showVorood, setShowVorood] = useState(true);
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/components/Header.tsx:8]: done

File: front/src/components/Header.tsx [hunk 5]
```diff
@@ -26,9 +26,33 @@ export default function Header() {
...
+        >
           ورود
         </Link>
+        <Link
+          to={"/store/cart"}
+          className={`text-[#58423A] text-lg z-[1] absolute right-32 top-[-12px] ${
+            showVorood ? "" : "hidden"
+          }`}
+        >
+          <svg
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/components/Header.tsx:43]: done

File: front/src/features/VideoPlayer/components/SessionList/SessionItem.tsx [hunk 6]
```diff
@@ -3,35 +3,57 @@ import { SessionItemProps } from "./types";
...
+        className={` text-sm text-[#fef3e9] opacity-70`}
+      >{`جلسه ${number2PersianOrdinal(props.data.num - 1)}`}</p>
+      <p className=" text-lg text-[#fef3e9] mt-auto mb-2">{props.data.title}</p>
+      <div className="flex justify-between items-center border-t-2 border-[#fef3e9b6] pt-2">
+        <p className="text-[#fef3e9b6] text-sm">{"description"}</p> {/* it probably needs text shorter */}
+        <div className="flex gap-1 items-center">
+          <p className="text-[#fef3e9b6] text-sm flex-row gap-2">
+            {"time"}
+          </p>
+          <svg
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/features/VideoPlayer/components/SessionList/SessionItem.tsx:41]: done

File: front/src/features/VideoPlayer/components/SessionList/index.tsx [hunk 7]
```diff
@@ -25,17 +23,19 @@ const SessionList: FC<WithTerm> = (props) => {
       ) : term.isLoading ? (
         <Loading textColor="light"/>
       ) : (
-        <>
-          {term.data.course.videos.length ? (
+        <div className="flex flex-col gap-4 w-full">
+          <SessionItem data={term.data.course.videos[0]} key={`video-${term.data.course.videos[0].id}`} />
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/features/VideoPlayer/components/SessionList/index.tsx:27]: done

File: front/src/features/VideoPlayer/components/TermController/RemainedDays.tsx [hunk 8]
```diff
@@ -2,25 +2,34 @@ import classNames from "classnames";
...
+        "..."
+      ) : (
+        <div className="flex gap-1">
+          <p className="text-sm">
+            {English2Persian(
+              `${isoStringDateDiffFromNow(term.data.course.end_date)} روز`
+            )}
+          </p>
+          <p className="text-sm text-[#58423A50]">
+            {`از دوره ${term.data.course.title} ${getJalaliMonthName(
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/features/VideoPlayer/components/TermController/RemainedDays.tsx:27]: done

File: front/src/pages/Store/Cart/component/FactorItem.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import React, { FC } from "react";
+import { IFactorItem } from "../types";
+import { getJalaliMonthName } from "src/utils/dates";
+import { level2farsi } from "src/utils/shopTranslator";
+import {
+  English2Persian,
+  addCommaEach3Digits,
+  addToman,
+} from "src/utils/converts";
+const FactorItem: React.FC<IFactorItem> = ({ level, month, price, title }) => {
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Cart/component/FactorItem.tsx:10]: done

File: front/src/pages/Store/Cart/component/FactorItem.tsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        ) : (
+          <h5>{title}</h5>
+        )}
+        <span className="text-[#58423A] text-base "> | </span>
+        <h5 className="text-[#58423a80] text-base ">
+          {getJalaliMonthName(month)} ماه
+        </h5>
+      </div>
+      <h5 className="text-[#58423A] text-base ">
+        {addToman(English2Persian(addCommaEach3Digits(price)))}
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Cart/component/FactorItem.tsx:25]: done

File: front/src/pages/Store/Cart/index.tsx [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+const Cart: React.FC = ({}) => {
+  const navigate = useNavigate();
+  const { cartData, isCartError, isCartLoading, isCartSuccess } =
+    useContext(StoreContext);
+
+  const totalOverallPrice = (cartItems: ICartItem[]): number => {
+    return cartItems.reduce((total, item) => total + item.overallPrice, 0);
+  };
+
+  const submitOrder = () => {
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Cart/index.tsx:24]: done

File: front/src/pages/Store/Cart/index.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+import { SUBMIT_ORDER_URL } from "../api.data";
+import api from "src/services";
+import { redirect, useNavigate } from "react-router-dom";
+
+const Cart: React.FC = ({}) => {
+  const navigate = useNavigate();
+  const { cartData, isCartError, isCartLoading, isCartSuccess } =
+    useContext(StoreContext);
+
+  const totalOverallPrice = (cartItems: ICartItem[]): number => {
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Cart/index.tsx:20]: done

File: front/src/pages/Store/Cart/index.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    return cartItems.reduce((total, item) => total + item.overallPrice, 0);
+  };
+
+  const submitOrder = () => {
+    const res = api.post<any, ISubmitOrderRes>(SUBMIT_ORDER_URL)({});
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Cart/index.tsx:25]: done

File: front/src/pages/Store/Shop/components/ShopCourseCard.tsx [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+import { level2farsi } from "src/utils/shopTranslator";
+import { IShopCourseCard } from "../types";
+import api from "src/services";
+import { SHOP_ADD_ITEM_URL, SHOP_DELETE_ITEM_URL } from "../../api.data";
+import { useMutation } from "@tanstack/react-query";
+
+const ShopCourseCard: React.FC<IShopCourseCard> = ({
+  title,
+  price,
+  BGthumbURL,
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/Shop/components/ShopCourseCard.tsx:16]: done

File: front/src/pages/Store/StoreContext.tsx [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+    retry: 1,
+  });
+
+  const {
+    data: shopData,
+    isError: isShopError,
+    isLoading: isShopLoading,
+    isSuccess: isShopSuccess,
+  } = useQuery({
+    queryKey: ["shop"],
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/StoreContext.tsx:58]: done

File: front/src/pages/Store/index.tsx [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+  const location = useLocation();
+  let page: ReactNode;
+  if (location.pathname === "/store/shop") {
+    page = <Shop />;
+  }
+  if (location.pathname === "/store/cart") {
+    page = <Cart />;
+  }
+
+  return <StoreContextProvider>{page}</StoreContextProvider>;
```
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/pages/Store/index.tsx:17]: done

File: front/src/App.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (arsamhb) [reply front/src/App.tsx:42]: done

File: front/src/pages/Store/Shop/index.tsx (hunk 2 above)
Comment (arsamhb) [reply front/src/pages/Store/Shop/index.tsx:92]: done

**Pull request merged**